### ISSUE TITLE
kpt fn runs functions as nobody which is a hassle for scratch-based containers

### ISSUE THREAD

**Issue by johnbelamaric** [GoogleContainerTools/kpt#1628]
### Expected behavior
I can use scratch-based containers without adding a fake `/etc/passwd` file.

### Actual behavior
When running `kpt fn run`, `kpt` defaults to the user nobody. This means that when I create a container image from scratch, I have to create a fake `/etc/passwd` file with nobody in it, or the function execution will fail. This is not the end of the world but annoying. Wouldn't a numerical user ID be better?

### Steps to reproduce the behavior

Build a docker container from scratch, copying in your `kpt` function binary.
Run `kpt fn run` with a declarative function configuration pointing to that container.

**Comment by frankfarzan** [comment]
Is your request to have the kpt orchestrator run all containers with a well-known uid instead of user `nobody`?

If so, this would be a backwards incompatible change. Although the Functions Spec doesn't explicitly state user `nobody`, the API has been effectively defined via current implementation, so we need a very strong justification for changing it:

https://github.com/kubernetes-sigs/kustomize/blob/master/cmd/config/docs/api-conventions/functions-spec.md

I see two AI(s) here:

1. Through a combination of  functions SDK and documentation, make it easy to create container images that can be run with user `nobody`.
2. Functions Spec must be explicit about how the orchestrators invokes the container.

**Comment by johnbelamaric** [comment]
Yes, that is essentially the request. I suppose in my pipeline I can use
the --as-current-user flag. So it's not a huge deal, more of an
inconvenience. It seems an odd default to assume a specific username exists
in the container.
